gitbook/network-ftp.md

```diff
@@ -2,189 +2,186 @@
 ```
 
-The status code can be used to check whether the command was successful or failed: Codes lower than 400 represent success, all others represent errors. You can use the `isOk()` function as a shortcut to test a status code for success. 
+The status code can be used to check whether the command was successful or failed: Codes lower than 400 represent success, all others represent errors. You can use the `ok?` function as a shortcut to test a status code for success.
 
-```
-sf::Ftp::Response response = ftp.login();
-if (response.isOk())
-{
-    // success!
-}
+```ruby
+response = ftp.login("username", "password")
+if response.ok?
+    # success!
 else
-{
-    // error...
-}
+    # error...
+end
 ```
 
-If you don't care about the details of the response, you can check for success with even less code: 
+If you don't care about the details of the response, you can check for success with even less code:
 
-```
-if (ftp.login().isOk())
-{
-    // success!
-}
+```ruby
+if ftp.login("username", "password").ok?
+    # success!
 else
-{
-    // error...
-}
+    # error...
+end
 ```
 
-For readability, these checks won't be performed in the following examples in this tutorial. Don't forget to perform them in your code! 
+For readability, these checks won't be performed in the following examples in this tutorial. Don't forget to perform them in your code!
 
-Now that you understand how the class works, let's have a look at what it can do. 
+Now that you understand how the class works, let's have a look at what it can do.
 
 ## Connecting to the FTP server
 
-The first thing to do is connect to an FTP server. 
+The first thing to do is connect to an FTP server.
 
-```
-sf::Ftp ftp;
-ftp.connect("ftp.myserver.org");
+```ruby
+ftp = SF::Ftp.new
+ftp.connect("ftp.myserver.org")
 ```
 
-The server address can be any valid [IpAddress]({{book.api}}/IpAddress.html): A URL, an IP address, a network name, ... 
+The server address can be any valid [IpAddress]({{book.api}}/IpAddress.html): A URL, an IP address, a network name, ...
 
-The standard port for FTP is 21. If, for some reason, your server uses a different port, you can specify it as an additional argument: 
+The standard port for FTP is 21. If, for some reason, your server uses a different port, you can specify it as an additional argument:
 
-```
-sf::Ftp ftp;
-ftp.connect("ftp.myserver.org", 45000);
+```ruby
+ftp = SF::Ftp.new
+ftp.connect("ftp.myserver.org", 45000)
 ```
 
-You can also pass a third parameter, which is a time out value. This prevents you from having to wait forever (or at least a very long time) if the server doesn't respond. 
+You can also pass a third parameter, which is a time out value. This prevents you from having to wait forever (or at least a very long time) if the server doesn't respond.
 
-```
-sf::Ftp ftp;
-ftp.connect("ftp.myserver.org", 21, sf::seconds(5));
+```ruby
+ftp = SF::Ftp.new
+ftp.connect("ftp.myserver.org", 21, SF.seconds(5))
```

Comment:
SF.ip_address("ftp.myserver.org")
This works in C++ due to implicit conversion from string to ipaddress.............